### ISSUE TITLE
Feature request: ability to edit an entry

### ISSUE THREAD

**Issue by gapan** [rhinstaller/efibootmgr#49]
It would be nice if efibootmgr could edit the label and/or the loader for an existing entry. So that:
```
efibootmgr -b 1 -L "New label"
```
and
```
efibootmgr -b 1 -l "\\EFI\\new.efi"
```
would work.

**Comment by celesteking** [comment]
+1. Man page doesn't forbid doing that. Either put a warning or implement the feature.

**Comment by pellettiero** [comment]
+1

Great idea, it just would need to copy the efivar, edit it and reapply it, like a copy and paste (assuming that's how efi variables work)

**Comment by cbarrick** [comment]
I was actually expecting entries to be updated, e.g. to change kernel parameters for an existing entry. The output of the command gives no indication that this is not happening. (Though running successfully without root privileges is what ultimately gave it away as not doing anything.)

Even if editing entries is not to be supported, there should at least be an error thrown when a user tries to do this.

For now, I'm scripting out a delete-create-reorder routine as a work around.

**Comment by jarauh** [comment]
> +1. Man page doesn't forbid doing that. Either put a warning or implement the feature.

Yes, it would be good if the man page and the --help were more explicit about this.  The man page seems to be created by Debian, though, I'm not sure whom to contact there.

**Comment by mercuriete** [comment]
I have another idea:

dump the content of an entry and ouput it on a format that is a command ready for creation a new entry that is a copy of that dump.
So you can copy that command, edit a new information you need and create a new entry starting with the information from the dump.

I know this is not a good idea, but we need something first to copy and modify.


PS: I have a gentoo installation using uefi and starting the kernel directly without bootloader.
So my usecase is:
anytime I need to add a kernel argument I need to edit the entry adding in unicode at the end of the actual kernel args.
So for me is better to start windows and use easyuefi that have the capability of edit entries.

**Comment by maxicarlos08** [comment]
Why isn't this feature still not here? It would be so much better than re-creating the entry every time if you only want to add an kernel argument

**Comment by frozencemetery** [comment]
> Why isn't this feature still not here?

Hey, this isn't how we talk to each other.  Please respect that we're all people with lives etc..

I'm glad you're interested in the project, but as with all open source, the reason $thing hasn't been done is because no one's done it yet - and you're empowered to send patches should you wish to.
